### PR TITLE
Issue #5004: pitest: increase mutation coverage for pitest-checks-imp…

### PR DESCRIPTION
**Pull request by djydewang**
…orts profile to 100%

Issue #5004

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java [hunk 1]
```diff
@@ -636,7 +635,7 @@ else if (customImportOrderRules.contains(SAME_PACKAGE_RULE_GROUP)) {
                 }
                 if (SPECIAL_IMPORTS_RULE_GROUP.equals(group)) {
                     bestMatch = findBetterPatternMatch(importPath,
-                            SPECIAL_IMPORTS_RULE_GROUP, specialImportsRegExp, bestMatch);
+                            group, specialImportsRegExp, bestMatch);
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java]: Can you explain why this change is needed for pitest?
report didn't mention anything about this, only the if condition. 
If it is not needed, please restore the original.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java [hunk 2]
```diff
@@ -691,7 +690,7 @@ private static int compareImports(String import1, String import2) {
         final String separator = "\\.";
         final String[] import1Tokens = import1.split(separator);
         final String[] import2Tokens = import2.split(separator);
-        for (int i = 0; i < import1Tokens.length && i != import2Tokens.length; i++) {
+        for (int i = 0; i != import1Tokens.length && i != import2Tokens.length; i++) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java]: If we allow this, then imports with un-even lengths will be an issue, right?

One example is `java.util.Map` and `java.util.Map.Entry`.
The 2 are equal in every position, except one has `.Entry` on the end, which makes it have 1 greater length then the other.
The method will see the 2 imports as completely equal which is incorrect.

Can we add something after the `for` loop like:
````
if (result == 0 && import1Tokens.length != import2Tokens.length) {
    result = Integer.compare(import1Tokens.length, import2Tokens.length);
}
````

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java]: Example case:
````
$ cat TestClass.java
import java.util.Map.Entry;
import java.util.Map;

import com.B;
import com.A;

public class TestClass {
    void method() {
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="CustomImportOrderCheck">
<property name="sortImportsInGroupAlphabetically" value="true" />
<property name="specialImportsRegExp" value="com" />
<property name="customImportOrderRules" value="STANDARD_JAVA_PACKAGE###SPECIAL_IMPORTS" />
</module>
    </module>
</module>

$ java -jar checkstyle-8.5-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:5: Wrong lexicographical order for 'com.A' import. Should be before 'com.B'. [CustomImportOrder]
Audit done.
Checkstyle ends with 1 errors.
````

No violation on `java.util.Map.Entry`.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java (hunk 1 above)
Comment (djydewang) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java]: > report didn't mention anything about this, only the if condition.

Right, when we negate if condition and `group != SPECIAL_IMPORTS_RULE_GROUP`, `findBetterPatternMatch` method works right with `SPECIAL_IMPORTS_RULE_GROUP` paraｍeter which means if condition meaningless.
 But, obviously, the if condition can't be omitted. So I do this change.

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java]: > the if condition can't be omitted
> when we negate ... method works right with SPECIAL_IMPORTS_RULE_GROUP paraｍeter

There is no other way around this, like with a custom `specialImportsRegExp`?
`STANDARD_JAVA_PACKAGE_RULE_GROUP` doesn't have this issue and it has very similar code.

Comment (djydewang) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java]: > There is no other way around this

I have another way(add some tests) to fix it, but I think this code logic is better.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java (hunk 2 above)
Comment (djydewang) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java]: Done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java (hunk 1 above)
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java]: @romani What is your opinion?

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java]: I am fine with update.
